pkg/network/vmispec/interface.go

```diff
@@ -155,3 +155,18 @@ func FilterStatusInterfacesByNames(interfaces []v1.VirtualMachineInstanceNetwork
 
 	return filtered
 }
+
+// CopyInterfaceMacAddresses return a copy of the target interfaces slice where each interface MAC address is set from
+// the given source interfaces slice.
+func CopyInterfaceMacAddresses(srcIfaces []v1.Interface, targetIfaces []v1.Interface) []v1.Interface {
+	indexedSrcIfaces := IndexInterfaceSpecByName(srcIfaces)
+	var newIfaces []v1.Interface
+	for _, targetIface := range targetIfaces {
+		if srcIface, exist := indexedSrcIfaces[targetIface.Name]; exist {
+			ifaceCopy := targetIface.DeepCopy()
```

Comment:
Its not a must, we can mutate the interfaces in the given slice, but I got the impression we try to avoid this pattern (mutate object instead of recreating it)

Reply:
I depends on what case. You are pass to this method a copy of the VMI. I don't see a reason to copy the copy:)

Reply:
I removed this code following https://github.com/kubevirt/kubevirt/pull/9896/commits/6615177e30fbf7abf2942f54c6b65e7c5c948e96#r1244786510